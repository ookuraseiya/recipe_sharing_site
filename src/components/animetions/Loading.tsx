import ReactLoading from 'react-loading';

export const Loading = () => {
  return (
    <>
      <div className="loading">
        <ReactLoading type="spinningBubbles" color="#5a320a" />
      </div>
    </>
  );
};
