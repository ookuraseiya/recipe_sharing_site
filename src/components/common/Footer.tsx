export const Footer = () => {
  return (
    <>
      <footer className="footer">
        <div className="footer__wrapper">
          <p className="footer__lead">
            Copyright © 2024 Ōkura home recipe sharing site
          </p>
        </div>
      </footer>
    </>
  );
};
