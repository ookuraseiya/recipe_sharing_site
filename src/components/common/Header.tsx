export const Header = () => {
  return (
    <>
      <header className="header">
        <div className="header__container">
          <div className="header__list">
            <img className="header__logo" src={'/images/logo.webp'} alt="" />
            <div className="header__button--position">
              <button className="header__button">ログアウト</button>
            </div>
          </div>
        </div>
      </header>
    </>
  );
};
