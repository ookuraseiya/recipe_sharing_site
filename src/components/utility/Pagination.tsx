import { Link } from 'react-router-dom';

type PaginationProps = {
  pageId: number;
  currentPage: number;
  paginationNumber: number;
};

export const Pagination = ({
  pageId,
  currentPage,
  paginationNumber,
}: PaginationProps) => {
  return (
    <>
      <section className="pagination">
        <ul className="pagination__list">
          <li
            className={
              pageId === 1 ? 'pagination__disabled' : 'pagination__link'
            }
          >
            <Link className="pagination__href" to={`/${pageId - 1}`}>
              &lt;
            </Link>
          </li>
          <li className="pagination__position">
            {currentPage} / {paginationNumber}
          </li>
          <li
            className={
              pageId === paginationNumber
                ? 'pagination__disabled'
                : 'pagination__link'
            }
          >
            <Link className="pagination__href" to={`/${pageId + 1}`}>
              &gt;
            </Link>
          </li>
        </ul>
      </section>
    </>
  );
};
