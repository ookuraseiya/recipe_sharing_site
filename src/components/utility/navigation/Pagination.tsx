export const Pagination = () => {
  return (
    <>
      <section className="pagination">
        <ul className="pagination__list">
          <li className="pagination__link">&lt;</li>
          <li className="pagination__position">2 / 20</li>
          <li className="pagination__link">&gt;</li>
        </ul>
      </section>
    </>
  );
};
