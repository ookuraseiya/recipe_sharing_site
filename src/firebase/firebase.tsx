import { initializeApp } from 'firebase/app';
import { getAuth } from 'firebase/auth';
import { getFirestore } from 'firebase/firestore';
import { getStorage } from 'firebase/storage';

const firebaseConfig = {
  apiKey: String(import.meta.env.VITE_FIREBASE_API_KEY),
  authDomain: String(import.meta.env.VITE_FIREBASE_AUTH_DOMAIN),
  projectId: String(import.meta.env.VITE_FIREBASE_PROJECT_ID),
  storageBucket: String(import.meta.env.VITE_FIREBASE_STORAGE_BUCKET),
  messagingSenderId: String(import.meta.env.VITE_FIREBASE_MESSAGE_SENDER_ID),
  appId: String(import.meta.env.VITE_FIREBASE_APP_ID),

  // apiKey: String(process.env.VITE_FIREBASE_API_KEY),
  // authDomain: String(process.env.VITE_FIREBASE_AUTH_DOMAIN),
  // projectId: String(process.env.VITE_FIREBASE_PROJECT_ID),
  // storageBucket: String(process.env.VITE_FIREBASE_STORAGE_BUCKET),
  // messagingSenderId: String(process.env.VITE_FIREBASE_MESSAGE_SENDER_ID),
  // appId: String(process.env.VITE_FIREBASE_APP_ID),
};

const app = initializeApp(firebaseConfig);
export const storage = getStorage(app);
export const auth = getAuth(app);
export const db = getFirestore(app);
