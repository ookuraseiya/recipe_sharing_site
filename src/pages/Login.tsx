import { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { auth } from '../firebase/firebase';
import { signInWithEmailAndPassword } from 'firebase/auth';

export const Login = () => {
  const navigate = useNavigate();
  const [loginError, setLoginError] = useState<string>('');
  const [email, setEmail] = useState<string>('');
  const [password, setPassword] = useState<string>('');

  const handleLogin = async () => {
    if (email && password) {
      try {
        await signInWithEmailAndPassword(auth, email, password);
        setLoginError('');
        navigate('/1');
      } catch (error: unknown) {
        if (error instanceof Error) {
          setLoginError('メールアドレスかパスワードが違います。');
        }
      }
    } else {
      setLoginError('メールアドレスとパスワード両方入力してください。');
    }
  };

  return (
    <>
      <section className="login">
        <div className="login__container">
          <div className="login__error--position">
            <h1 className="login__error" data-testid="error-text">
              {loginError && loginError}
            </h1>
          </div>
          <form className="login__form" onSubmit={(e) => e.preventDefault()}>
            <h1 className="login__title">ログインフォーム</h1>
            <div className="login__form--group">
              <label className="login__form--label" htmlFor="email">
                Email:
              </label>
              <input
                className="login__form--input"
                type="email"
                id="email"
                name="email"
                value={email}
                onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                  setEmail(e.target.value)
                }
                required
              />
            </div>
            <div className="login__form--group">
              <label className="login__form--label" htmlFor="password">
                Password:
              </label>
              <input
                className="login__form--input"
                type="password"
                id="password"
                name="password"
                value={password}
                onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                  setPassword(e.target.value)
                }
                required
              />
            </div>
            <div className="login__button--position">
              <button
                className="login__button"
                type="submit"
                onClick={handleLogin}
              >
                Login
              </button>
            </div>
          </form>
        </div>
      </section>
    </>
  );
};
