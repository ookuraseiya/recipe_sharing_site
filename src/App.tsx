import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { Main } from './pages/Main';
import { RecipeDetail } from './pages/RecipeDetail';
import './css/styles.css';

function App() {
  return (
    <>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Main />} />
          <Route path="/recipeDetail/:recipeID" element={<RecipeDetail />} />
        </Routes>
      </BrowserRouter>
    </>
  );
}

export default App;
