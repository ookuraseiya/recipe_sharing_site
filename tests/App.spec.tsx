process.env.VITE_FIREBASE_API_KEY;
process.env.VITE_FIREBASE_AUTH_DOMAIN;
process.env.VITE_FIREBASE_PROJECT_ID;
process.env.VITE_FIREBASE_STORAGE_BUCKET;
process.env.VITE_FIREBASE_MESSAGE_SENDER_ID;
process.env.VITE_FIREBASE_APP_ID;

import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import App from '../src/App';
import { Header } from '../src/components/common/Header';
import { Footer } from '../src/components/common/Footer';
import { Login } from '../src/pages/Login';
import fetchMock from 'jest-fetch-mock';
import { signInWithEmailAndPassword } from 'firebase/auth';
import { Loading } from '../src/components/animetions/Loading';
import { Pagination } from '../src/components/utility/Pagination';

fetchMock.enableMocks();

// Firebaseのモジュールをモック化
jest.mock('firebase/auth', () => ({
  ...jest.requireActual('firebase/auth'),
  signInWithEmailAndPassword: jest.fn((email: string, password: string) => {
    if (email === process.env.EMAIL && password === process.env.PASSWORD) {
      return Promise.resolve();
    } else {
      return Promise.reject(
        new Error('メールアドレスかパスワードが違います。')
      );
    }
  }),
}));

// scrollTopのモック化
global.window.scrollTo = jest.fn();

// Appコンポーネントのテスト;
describe('Appコンポーネントのテスト', () => {
  test('Appページのレンダーテスト', () => {
    render(<App />);
  });

  test('Appコンポーネントのスナップショットテスト', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});

// Loginコンポーネントのテスト;
describe('Loginコンポーネントのテスト', () => {
  test('正しいメールアドレスと正しいパスワードを入力したときのテスト', async () => {
    // モック関数をセットアップ
    const mockSignIn = signInWithEmailAndPassword as jest.MockedFunction<
      typeof signInWithEmailAndPassword
    >;

    const email = process.env.EMAIL;
    const password = process.env.PASSWORD;
    render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    );

    fireEvent.change(screen.getByLabelText('Email:'), {
      target: { value: email },
    });
    fireEvent.change(screen.getByLabelText('Password:'), {
      target: { value: password },
    });

    fireEvent.click(screen.getByText('Login'));

    await waitFor(() => {
      expect(mockSignIn).toHaveBeenCalledWith(
        expect.anything(),
        email,
        password
      );
      expect(screen.getByTestId('error-text')).toBeEmptyDOMElement();
    });
  });

  test('正しくないメールアドレスと正しいパスワードを入力したときのテスト', async () => {
    // モック関数をセットアップ
    const mockSignIn = signInWithEmailAndPassword as jest.MockedFunction<
      typeof signInWithEmailAndPassword
    >;

    const email = 'test@gmail.com';
    const password = process.env.PASSWORD;
    render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    );

    fireEvent.change(screen.getByLabelText('Email:'), {
      target: { value: email },
    });
    fireEvent.change(screen.getByLabelText('Password:'), {
      target: { value: password },
    });

    fireEvent.click(screen.getByText('Login'));

    await waitFor(() => {
      expect(mockSignIn).toHaveBeenCalledWith(
        expect.anything(),
        email,
        password
      );
      expect(
        screen.getByText('メールアドレスかパスワードが違います。')
      ).toBeInTheDocument();
    });
  });

  test('正しいメールアドレスと正しくないパスワードを入力したときのテスト', async () => {
    // モック関数をセットアップ
    const mockSignIn = signInWithEmailAndPassword as jest.MockedFunction<
      typeof signInWithEmailAndPassword
    >;

    const email = process.env.EMAIL;
    const password = 'dummyPassword';
    render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    );

    fireEvent.change(screen.getByLabelText('Email:'), {
      target: { value: email },
    });
    fireEvent.change(screen.getByLabelText('Password:'), {
      target: { value: password },
    });

    fireEvent.click(screen.getByText('Login'));

    await waitFor(() => {
      expect(mockSignIn).toHaveBeenCalledWith(
        expect.anything(),
        email,
        password
      );
      expect(
        screen.getByText('メールアドレスかパスワードが違います。')
      ).toBeInTheDocument();
    });
  });

  test('メールアドレスが空白でパスワードが入力されたときのテスト', async () => {
    const email = '';
    const password = process.env.PASSWORD;
    render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    );

    fireEvent.change(screen.getByLabelText('Email:'), {
      target: { value: email },
    });
    fireEvent.change(screen.getByLabelText('Password:'), {
      target: { value: password },
    });

    fireEvent.click(screen.getByText('Login'));

    await waitFor(() => {
      expect(
        screen.getByText('メールアドレスとパスワード両方入力してください。')
      ).toBeInTheDocument();
    });
  });

  test('メールアドレスが入力されて、パスワードが空白のときのテスト', async () => {
    const email = process.env.EMAIL;
    const password = '';
    render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    );

    fireEvent.change(screen.getByLabelText('Email:'), {
      target: { value: email },
    });
    fireEvent.change(screen.getByLabelText('Password:'), {
      target: { value: password },
    });

    fireEvent.click(screen.getByText('Login'));

    await waitFor(() => {
      expect(
        screen.getByText('メールアドレスとパスワード両方入力してください。')
      ).toBeInTheDocument();
    });
  });
});

// Headerコンポーネントのテスト
describe('Headerコンポーネントのテスト', () => {
  test('Headerコンポーネントのレンダーテスト', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
  });

  test('Headerページのスナップショットテスト', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

// Loadingコンポーネントのテスト
describe('Loadingコンポーネントのテスト', () => {
  test('Loadingのレンダーテスト', () => {
    render(
      <BrowserRouter>
        <Loading />
      </BrowserRouter>
    );
  });
});

// Paginationコンポーネントのテスト
describe('Paginationコンポーネントのテスト', () => {
  test('Paginationのレンダーテスト', () => {
    render(
      <BrowserRouter>
        <Pagination pageId={1} currentPage={1} paginationNumber={10} />
      </BrowserRouter>
    );
  });

  test('Paginationコンポーネントのスナップショットテスト', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Pagination pageId={1} currentPage={1} paginationNumber={10} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('Paginationの子コンポーネントのレンダーテスト', () => {
    render(
      <BrowserRouter>
        <Pagination pageId={1} currentPage={1} paginationNumber={10} />
      </BrowserRouter>
    );
    expect(screen.getByText('<')).toBeInTheDocument();
    expect(screen.getByText('>')).toBeInTheDocument();
    expect(screen.getByText('1 / 10')).toBeInTheDocument();
  });

  test('Paginationのパステスト', () => {
    render(
      <BrowserRouter>
        <Pagination pageId={1} currentPage={1} paginationNumber={10} />
      </BrowserRouter>
    );
    expect(screen.getByText('<').closest('a')).toHaveAttribute('href', '/0');
    expect(screen.getByText('>').closest('a')).toHaveAttribute('href', '/2');
  });

  test('PaginationコンポーネントにLinkが2つあるか', () => {
    const { getAllByRole } = render(
      <BrowserRouter>
        <Pagination pageId={1} currentPage={1} paginationNumber={10} />
      </BrowserRouter>
    );
    const links = getAllByRole('link');
    expect(links.length).toBe(2);
  });
});

// Footerコンポーネントのテスト
describe('Footerコンポーネントのテスト', () => {
  test('Footerコンポーネントのレンダーテスト', () => {
    render(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>
    );
  });

  test('Footerページのスナップショットテスト', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
