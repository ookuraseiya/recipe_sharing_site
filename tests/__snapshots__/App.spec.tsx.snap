// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appコンポーネントのテスト Appコンポーネントのスナップショットテスト 1`] = `<DocumentFragment />`;

exports[`Appコンポーネントのテスト Headerページのスナップショットテスト 1`] = `<DocumentFragment />`;

exports[`Footerコンポーネントのテスト Footerページのスナップショットテスト 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="footer__wrapper"
    >
      <p
        class="footer__lead"
      >
        Copyright © 2024 Ōkura home recipe sharing site
      </p>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Headerコンポーネントのテスト Headerページのスナップショットテスト 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="header__container"
    >
      <div
        class="header__list"
      >
        <img
          alt=""
          class="header__logo"
          src="/images/logo.webp"
        />
        <div
          class="header__button--position"
        >
          <button
            class="header__button"
          >
            ログアウト
          </button>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Paginationコンポーネントのテスト Paginationコンポーネントのスナップショットテスト 1`] = `
<DocumentFragment>
  <section
    class="pagination"
  >
    <ul
      class="pagination__list"
    >
      <li
        class="pagination__disabled"
      >
        <a
          class="pagination__href"
          href="/0"
        >
          &lt;
        </a>
      </li>
      <li
        class="pagination__position"
      >
        1 / 10
      </li>
      <li
        class="pagination__link"
      >
        <a
          class="pagination__href"
          href="/2"
        >
          &gt;
        </a>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
